### PR TITLE
determine if plugins need to be updated during the execution …

### PR DESCRIPTION
**Pull request by sd-obermeyer**
…of AutomaticDeployCommand in order to be able to update plugins which depend on each other

### CODE REVIEW

File: src/Service/TableParser.php [hunk 1]
```diff
@@ -76,7 +87,7 @@ private function parseLine(string $line): array
         });
 
         if ($this->stripFirstCell) {
-            \array_shift($cells);
+            $test = \array_shift($cells);
```
Comment (sd-lueckel) [inline src/Service/TableParser.php:90]:
```suggestion
            \array_shift($cells);
```

File: src/Command/ListCommand.php [hunk 2]
```diff
@@ -42,11 +64,41 @@ protected function execute(
         InputInterface $input,
         OutputInterface $output
     ) {
-        // @TODO Read plugins from database
+        // @TODO Read plugins from database -> done
```
Comment (Jnoack331) [inline src/Command/ListCommand.php:67]: then remove the todo?!? x)

**Review by sd-lueckel — APPROVED** [review]

File: src/Command/ListCommand.php (hunk 2 above)
Comment (sd-obermeyer) [reply src/Command/ListCommand.php:67]: Dann würde bei den TODOs drunter aber Kontext fehlen 🙂

File: src/Service/PluginVersionService.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            $name = $pluginLine[0];
+            $version = $pluginLine[2];
+
+            $this->pluginVersions[$name] = $version;
+        }
+    }
+
+    public function getPluginVersion(string $pluginName): ?string
+    {
+        if (isset($this->pluginVersions[$pluginName])) {
```
Comment (twojtylak) [inline src/Service/PluginVersionService.php:61]: I don't know the details of the code, but I think you should either define `pluginsVersions` as empty array by default or check if pluginVersions is an array

Comment (sd-obermeyer) [reply src/Service/PluginVersionService.php:61]: The problem with this is that I want to make sure in the `pluginNeedsUpdate` method that the plugin list has been initialized by calling `parsePluginVersionsFromPluginList`. An empty array would however be a completely valid return value for the `parsePluginVersionsFromPluginList` method, thus making it unfeasible as the default value for the plugin list.

Comment (twojtylak) [reply src/Service/PluginVersionService.php:61]: How about an own property `initialized` instead of giving an existing one multiple meanings.

Comment (sd-obermeyer) [reply src/Service/PluginVersionService.php]: Yes, that is probably a better solution, I updated the PR accordingly 🙂

**Review by twojtylak — APPROVED** [review]

**Review by sd-lueckel — APPROVED** [review]

**Pull request merged**